### PR TITLE
update Ceph upgrade docs for 1.6

### LINKED ISSUE #7528: Update the Ceph upgrade guide for the 1.6 release
**Is this a bug report or feature request?**
* Feature Request

**What should the feature do:**
For the v1.6 release we need to update the upgrade guide and manually test the upgrades.

### PR DESCRIPTION
**Pull request by BlaineEXE**
Signed-off-by: Blaine Gardner <blaine.gardner@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #7528 

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

**Comment by BlaineEXE** [comment]
Notes on upgrade testing. Upgrade seems to happen successfully on the surface, but things are concerning a little more deeply. I tested upgrading from Rook v1.5.0 to the release-1.6 branch.

### double reconcile 
There is a double reconcile on the CephCluster, but I'm not sure yet what causes it

### dashboard cred failures
I see failures from setting dashboard creds.
```
2021-04-09 21:05:30.092159 E | op-mgr: failed modules: "dashboard". failed to initialize dashboard: failed to set login credentials for the ceph dashboard: failed to set login creds on mgr: failed to complete command for set dashboard creds: Invalid command: unused arguments: ['\\oT5"iY=H^$8)K#Oz.2V']
dashboard set-login-credentials <username> :  Set the login credentials. Password read from -i <file>
Traceback (most recent call last):
  File "/usr/bin/ceph", line 1310, in <module>
    retval = main()
  File "/usr/bin/ceph", line 1256, in main
    outf.write(outbuf)
TypeError: a bytes-like object is required, not 'str'
.
```

### Object store user failure
I see failures updating object store users. The `CephObjectStoreUser` goes into `ReconcileFailed` state.

File: Documentation/ceph-upgrade.md [hunk 1]
```diff
@@ -51,23 +52,23 @@ With this upgrade guide, there are a few notes to consider:
...
 another are as simple as updating the common resources and the image of the Rook operator. For
-example, when Rook v1.5.3 is released, the process of updating from v1.5.0 is as simple as running
+example, when Rook v1.6.1 is released, the process of updating from v1.6.0 is as simple as running
 the following:
 
-First get the latest common resources manifests that contain the latest changes for Rook v1.5.
+First get the latest common resources manifests that contain the latest changes for Rook v1.6.
 ```sh
-git clone --single-branch --branch v1.5.3 https://github.com/rook/rook.git
+git clone --single-branch --branch v1.6.0 https://github.com/rook/rook.git
```
Comment (travisn) [inline Documentation/ceph-upgrade.md:60]: The example branch tag should be the same as the target version
```suggestion
git clone --single-branch --branch v1.6.1 https://github.com/rook/rook.git
```

File: Documentation/ceph-upgrade.md [hunk 2]
```diff
@@ -399,15 +402,9 @@ The majority of the upgrade will be handled by the Rook operator. Begin the upgr
 Ceph image field in the cluster CRD (`spec.cephVersion.image`).
 
 ```sh
-<<<<<<< HEAD
 NEW_CEPH_IMAGE='ceph/ceph:v15.2.9-20210224'
 CLUSTER_NAME="$ROOK_CLUSTER_NAMESPACE"  # change if your cluster name is not the Rook namespace
 kubectl -n $ROOK_CLUSTER_NAMESPACE patch CephCluster $CLUSTER_NAME --type=merge -p "{\"spec\": {\"cephVersion\": {\"image\": \"$NEW_CEPH_IMAGE\"}}}"
-=======
-NEW_CEPH_IMAGE='ceph/ceph:v15.2.8-20201217'
```
Comment (travisn) [inline Documentation/ceph-upgrade.md:407]: hmm, where did this come from?

**Review by travisn — CHANGES_REQUESTED** [review]
Nice, there really isn't anything major with this upgrade process. It is theoretically as simple for the 1.6 upgrade as a patch upgrade. Just a tiny suggestion...

> ### double reconcile
> There is a double reconcile on the CephCluster, but I'm not sure yet what causes it

I didn't see a double reconcile. Did you perhaps update the rook version and ceph to pacific at the same time? That may cause a double reconcile.

> ### dashboard cred failures
> I see failures from setting dashboard creds.
This will be fixed by #7498, it is marked as a blocker for 1.6.

> 
> ### Object store user failure
> I see failures updating object store users. The `CephObjectStoreUser` goes into `ReconcileFailed` state.

I see this as well, could you open an issue?

File: Documentation/ceph-upgrade.md (hunk 2 above)
Comment (subhamkrai) [reply Documentation/ceph-upgrade.md:407]: @travisn I think my PR related to doc convention brought this.

File: Documentation/ceph-upgrade.md (hunk 1 above)
Comment (leseb) [reply Documentation/ceph-upgrade.md:60]: I think adding `--depth 1` is interesting too.

File: Documentation/ceph-upgrade.md [hunk 3]
```diff
@@ -276,23 +277,23 @@ needed by the Operator. Also update the Custom Resource Definitions (CRDs).
 > `rbac.authorization.k8s.io/v1beta1` instead of `rbac.authorization.k8s.io/v1`
 > You will also need to apply `pre-k8s-1.16/crds.yaml` instead of `crds.yaml`.
 
-First get the latest common resources manifests that contain the latest changes for Rook v1.5.
+First get the latest common resources manifests that contain the latest changes for Rook v1.6.
 ```sh
-git clone --single-branch --branch v1.5.3 https://github.com/rook/rook.git
+git clone --single-branch --branch v1.6.0 https://github.com/rook/rook.git
```
Comment (leseb) [inline Documentation/ceph-upgrade.md:282]: ditto?

File: Documentation/ceph-upgrade.md (hunk 1 above)
Comment (travisn) [reply Documentation/ceph-upgrade.md:60]: Since we're using `--single-branch`, that's already limiting it, right? Just in the interest of keeping the command simple.

Comment (leseb) [reply Documentation/ceph-upgrade.md:60]: Yes and no, we still pull the entire history of the branch.

**Comment by BlaineEXE** [comment]
@travisn I only saw the double reconcile once. I'm not sure what caused it, but maybe it was a fluke or something caused by ETCD cache invalidation.

File: Documentation/ceph-upgrade.md (hunk 1 above)
Comment (satoru-takeuchi) [reply Documentation/ceph-upgrade.md:60]: `git clone --depth 1 --branch v1.6.0 https://github.com/rook/rook.git` is enough. It's fast and simple.

```man 1 git-clone
       --depth <depth>
           Create a shallow clone with a history truncated to the specified number of commits. Implies --single-branch unless --no-single-branch is given to fetch the histories near the tips of all branches. If you want to clone submodules shallowly, also pass --shallow-submodules.
```

File: Documentation/ceph-upgrade.md [hunk 4]
```diff
@@ -51,23 +52,23 @@ With this upgrade guide, there are a few notes to consider:
...
 another are as simple as updating the common resources and the image of the Rook operator. For
-example, when Rook v1.5.3 is released, the process of updating from v1.5.0 is as simple as running
+example, when Rook v1.6.1 is released, the process of updating from v1.6.0 is as simple as running
 the following:
 
-First get the latest common resources manifests that contain the latest changes for Rook v1.5.
+First get the latest common resources manifests that contain the latest changes for Rook v1.6.
 ```sh
-git clone --single-branch --branch v1.5.3 https://github.com/rook/rook.git
+git clone --single-branch --depth=1 --branch v1.6.1 https://github.com/rook/rook.git
```
Comment (travisn) [inline Documentation/ceph-upgrade.md:60]: `depth=1` looks good, we should apply it in the docs as well where we use the `single-branch` flag, but we could do it in a separate PR to avoid the merge conflicts with the backport. Those lines have all been modified in the release branch.

**Review by travisn — APPROVED** [review]

**Review by satoru-takeuchi — APPROVED** [review]
